restore-workspace-tutorial-compiler/index.asciidoc

```diff
@@ -0,0 +1,138 @@
+= Restore a workspace
+== Learn how to use restoreWorkspace inside your tutorials
+
+====
+In this scenario, you will learn how to use the restoreWorkspace() function inside your tutorials.
+
+## Prerequisites
+* GitHub Account 
+* tutorial-compiler installed 
+
+If you haven't setup the tutorial-compiler yet, you can follow this link to learn it: 
+https://katacoda.com/devonfw/scenarios/create-your-own-tutorial.
+
+
+## Learning goals
+In this tutorial you will learn, how to: 
+* use 'restoreWorkspace' inside your tutorials
+* name the new workspace repository
+* clone a forked repository as workspace
+* use a local folder as workspace 
+
+You can find the documentation of this function here: 
+https://github.com/devonfw-tutorials/tutorials/wiki/Functions
+
+====
+
+[step]
+--
+addSetupScript("setup/setupTutorialCompiler.sh", "setup/setupTutorialCompiler.sh")
+--
+
+====
+
+Usually, the name of the workspace repository is a concatenation of "workspace-" and the name of your playbook-folder.
+In this tutorial, we will use this repository here:
+
+https://github.com/devonfw-tutorials/workspace-devon4ng
+
+Its name is the concatenation of 'workspace-' and 'devon4ng'.
+The runners will look for a tutorial folder, which is named 'devon4ng'. 
+
+[step]
+== Naming convention and usage
+--
+createFile("tutorials/devon4ng/index.asciidoc", "files/devon4ng-as-folder.asciidoc")
+nextKatacodaStep("Specify a different repository",  [{"file": "files/workspace-parameter.md"}])
+createFile("tutorials/devon4ng-architecture/index.asciidoc", "files/devon4ng-as-parameter.asciidoc")
+--
+
+Both ways will clone 
+
+https://github.com/devonfw-tutorials/workspace-devon4ng.git
+
+ into our workspace directory. 
+
+Now you know, which name to choose for your repository and how to use 'restoreWorkspace' inside your tutorials.
+
+A quick summary: 
+
+1) naming convention is the concatenation of 'workspace-' and '[name]'
+
+2) 'restoreWorkspace()'' will use the tutorial folder name as [name]
+
+3) 'restoreWorkspace({"workspace": "name"})' will use the parameter as [name]
+====
+====
+
+If you want to change the workspace repository, it might be useful to choose a forked repository as workspace to test your tutorial. 
+
+Before cloning the repository we have to fork it first.
+
+Go to the workspace repository on GitHub. In our tutorial it would be:
+
+https://github.com/devonfw-tutorials/workspace-devon4ng.git
+
+Fork the repository by clicking on the 'Fork' button.
+This will create a copy of the repository in your own GitHub account. You will find it in 'Your repositories'.
+
+[step]
+== Clone a forked repository as workspace
+--
+nextKatacodaStep("Fork the workspace- repository", [{"image": "images/fork.jpg"}, {"content": "\nNow you can run your tutorial and add your username and the right branch to the command.\n"}])
+--
+If you run 
+
+'localBuildRun.sh --user [YOUR_NAME] --branch [YOUR_BRANCH]'
```

Comment:
The user has to use 'bash' before localBuildRun.sh. And the user has to be in the directory of the tutorial-compiler